go/src/staleFlagCleaner.go

```diff
@@ -0,0 +1,624 @@
+		switch exprUnderConsideration.Op {
+		case token.NOT:
+			valX := sfc.evaluateExprNode(exprUnderConsideration.X)
+			switch valX {
+			case isTrue:
+				return isFalse
+			case isFalse:
+				return isTrue
+			}
+		case token.AND: // &expr
+			return sfc.evaluateExprNode(exprUnderConsideration.X)
+		default:
+			//do nothing for now.
+		}
+	//Star expression (*expr)
+	case *dst.StarExpr:
+		return sfc.evaluateExprNode(exprUnderConsideration.X)
+	//Paren Expr: exprs inside parenthesis
+	case *dst.ParenExpr:
+		return sfc.evaluateExprNode(exprUnderConsideration.X)
+
+	//Selector Expr (expr.A, a.b, etc.)
+	case *dst.SelectorExpr:
+		switch ident := exprUnderConsideration.X.(type) {
+		case *dst.Ident:
+			if val, ok := sfc.valueMap[ident.Name+"."+exprUnderConsideration.Sel.Name]; ok {
+				return val
+			}
+		default:
+			//Do nothing for now
+		}
+
+	default:
+		//Do nothing for now
+	}
+
+	return isBot
+}
+
+// This is the pre function of the dstutil.Apply
+/*
+From documentation:
+If pre is not nil, it is called for each node before the node's children are
+traversed (pre-order). If pre returns false, no children are traversed, and
+not called for that node.
+*/
+func (sfc *staleFlagCleaner) pre(n *dstutil.Cursor) bool {
+	return true
+}
+
+func (sfc *staleFlagCleaner) checkForBoolLiterals(value dst.Expr) bool {
+	switch checkval := value.(type) {
+	case *dst.Ident:
+		if checkval.Name == strTrue || checkval.Name == strFalse {
+			//do nothing
+			return true
+		}
+	default:
+		return false
+	}
+	return false
+}
+
+/*
+Below function have this aim:
+If Value is isBot or it is a bool literal, then we are not adding that to valueMap
+if not then will add to valueMap.
+This signifies whether variable has effect of flag or not.
+*/
+func (sfc *staleFlagCleaner) updateValueMapPost(name dst.Expr, value dst.Expr) {
+	valOfExpr := sfc.evaluateExprNode(value)
+	doNotDelLiterals := sfc.checkForBoolLiterals(value)
+	switch d := name.(type) {
+	case *dst.Ident:
+		if valOfExpr == isBot || doNotDelLiterals {
+			if _, ok := sfc.valueMap[d.Name]; ok {
+				delete(sfc.valueMap, d.Name)
+			}
+		} else {
+			sfc.valueMap[d.Name] = valOfExpr
+		}
+	case *dst.SelectorExpr:
+		switch ident := d.X.(type) {
+		case *dst.Ident:
+			if valOfExpr == isBot || doNotDelLiterals {
+				if _, ok := sfc.valueMap[ident.Name+"."+d.Sel.Name]; ok {
+					delete(sfc.valueMap, ident.Name+"."+d.Sel.Name)
+				}
+			} else {
+				sfc.valueMap[ident.Name+"."+d.Sel.Name] = valOfExpr
+			}
+		default:
+			//Do nothing for now
+		}
+	default:
+		// other cases may come here in future
+	}
+}
+
+/*
+It will deal with *dst.ValueSpec. Values of variable stored in valueMap
+will be deleted from node.
+*/
+func (sfc *staleFlagCleaner) DelValStmt(names *[]*dst.Ident, values *[]dst.Expr) bool {
+	valOfExpr := isBot
+	delCallExpr := false
+	doNotDelLiterals := false
+	for ind := 0; ind < len((*names)); ind++ {
+		delCallExpr = false
+		doNotDelLiterals = false
+		if len((*values)) != 0 {
```

Comment:
This mainly handles these type of expressions
```
var x, y bool = false, false
var c, d bool
```
We don't need to check `len((*names)) == len((*values))` because they will be always same if values are initialised. Go usually ask to assign all variables or no variables in this particular fashion.

Reply:
I haven't handed these expressions explicitly.
`a, b := foo()`

For these types of functions, we need to fix a convention on which variable need to be considered for refactoring.